packages/plugin-core/src/WSUtilsV2.ts

```diff
@@ -95,6 +96,48 @@ export class WSUtilsV2 implements IWSUtilsV2 {
     });
   }
 
+  /**
+   * See {@link IWSUtilsV2.getNoteFromMultiVault}
+   */
+  async getNoteFromMultiVault(opts: { fname: string; vault?: DVault }) {
```

Comment:
Why is this duplicated here when we have `NoteUtils.getNoteFromMultiVault` already? Can we refactor this to share the same code? Code duplication will make maintenance significanly harder in the future.

Reply:
This one is different from the `NoteUtils` one as this one will prompt the user to pick the vault for multiple matches. Long term I'd like to deprecate the `NoteUtils` version to use this non-static version but I'll like to address that in a later refactoring

Reply:
Please mark `NoteUtils.getNoteFromMultiVault` as deprecated then (`@deprecated, see {@link WSUtils.getNoteFromMultiVault}`.

Reply:
Sure, I'll deprecate after I start looking into refactoring which will be in another pr. I'll leave a TODO note here to remind myself

Reply:
Please just mark it as deprecated now. I'm not asking you to investigate all the usages of it or anything like that, just mark it as deprecated with a comment so nobody else uses it in the future. That's the same thing as putting a TODO there, but this way VSCode will show a warning in the editor so people now not to use it.

```ts
/** @deprecated see {@link WSUtils.getNoteFromMultiVault} */
async getNoteFromMultiVault() {
```

Reply:
So on second look, these two methods are actually quite different. The `NoteUtils` one gets a list of notes by fname and picks one that matches the current vault whereas the `WSUtils` one gets a list of notes by fname and prompts user to pick one. Very different use cases. They also don't have much overlapping code that is worth refactoring out.

I can rename mine so it doesn't get confusing

Reply:
I think you could refactor it into a common function that accepts a callback to resolve the "more than 1 maybe note" case, but it's fine if you'd rather deal with that later.

Reply:
That sounds like a good idea. I'll take a look into it when I try to refactor the code later